### PR TITLE
Remove building Docker containers with MariaDB

### PR DESCRIPTION
**Pull request by dbussink**
In https://github.com/vitessio/vitess/issues/9518 MariaDB support was deprecated in v14 and is being EOL'd in v16, which means that at this point for the future Vitess v16 we don't need to publish containers with MariaDB anymore.

## Related Issue(s)

In https://github.com/vitessio/vitess/issues/9518 we deprecated MariaDB support in v14 and are EOLing it in v16 so we should stop publishing artifacts with it in v16 as well. 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by mattlord** [comment]
I personally like it. I intentionally left these parts alone in https://github.com/vitessio/vitess/pull/12036 as it was not directly related to the issue being addressed in that PR and more importantly I was not sure what knock on effects this might have or what other related work may be needed. I'll let @frouioui and @deepthi comment on that and let them formally approve.

**Review by frouioui — APPROVED** [review]

**Comment by frouioui** [comment]
Once this Pull Request is merged we will remove the auto build for `mariadb` and `mariadb103` Docker tags.

**Review by deepthi — COMMENTED** [review]
What is the reason to remove the alpine docker file? Did you look at the history to see why it was added?

**Review by mattlord — APPROVED** [review]

**Pull request merged**
